Automattic/sensei
# Add extensions layout fetch

### Changes proposed in this Pull Request

* It updates the extensions layout data mock to fetch the real API.
* The solution basically fetches it through the server, saving it in a transient for 1 day. So we can avoid many requests to senseilms.com, and we can easily prepare it to preload using the same approach we used in other fetches.

### Testing instructions



~~It depends on 81-gh-wpcomvip/a8c-senseilms-com, so if it's not merged yet, you can checkout that branch locally, and update the `Sensei_Extensions::SENSEILMS_PRODUCTS_API_BASE_URL` to your local URL. In this case, make sure your server can see the the API server.~~

* Enable the feature flag: `define( 'SENSEI_FEATURE_FLAG_EXTENSIONS_MANAGEMENT_ENHANCEMENT', true );`
* Access the extensions page, and make sure it works properly fetching the API.
* If you wanna test different scenarios changing the API, remember to erase the `sensei_extensions_layout` transient.
